### PR TITLE
[Process] document command-as-arrays

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixed https://github.com/symfony/symfony-docs/issues/9897

### CODE REVIEW

File: components/process.rst [hunk 1]
```diff
@@ -100,6 +100,33 @@ with a non-zero code)::
...
+
+    .. versionadded:: 3.3
+        The ability to define commands as array of arguments was introduced in Symfony 3.3.
+
+    When defining commands as arrays, its arguments are escaped for you and you should not
+    escape them any further. You can also define commands as strings. When doing so, the
+    command-line is parsed by the underlying shell of your operating system. This allows
+    using e.g. stream redirections or conditional execution, but this is a lot less portable
+    since each OS (esp. Windows vs Unix-like) deals with escaping and parsing differently.
+    When passing a command-line string, it becomes your responsibility to deal this.
```
Comment (HeahDude) [inline components/process.rst]: `deal with this`

**Review by HeahDude — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

File: components/process.rst [hunk 2]
```diff
@@ -100,6 +100,33 @@ with a non-zero code)::
...
+    ``start()`` methods. Referencing them in command-line strings is OS-dependent::
+
+        // On Unix-like OSes
+        $process = new Process('echo "$MESSAGE"');
+
+        // On Windows
+        $process = new Process('echo %MESSAGE%');
+
+        // On both Unix-like and Windows
+        $process->run(null, 'MESSAGE' => 'Something to output']);
```
Comment (Toflar) [inline components/process.rst]: The command is missing or what did I miss?

Comment (javiereguiluz) [reply components/process.rst]: The command is `echo ...` This line is the `->run()` method call, which is the same for all OS. Only the command definition depends on the OS.

Comment (Toflar) [reply components/process.rst]: I see, when I read this part it feels like it wants to show me the difference between using a string or an array value which is not the case. That's a bit confusing. 
Not sure you understand what I mean. So when I read `On both Unix-like and Windows` it sounds to me like the example should tell me how I could make the command run on both systems without dealing with escaping. Oh man, this is hard to explain 😄

File: components/process.rst [hunk 3]
```diff
@@ -100,6 +100,32 @@ with a non-zero code)::
...
     }
 
+.. tip::
+
+    .. versionadded:: 3.3
+        The ability to define commands as arrays of arguments was introduced in
+        Symfony 3.3.
+
+    Defining commands as arrays of arguments is recommended because the Process
+    component escapes those arguments automatically. Defining commands as
```
Comment (Toflar) [inline components/process.rst]: "Automatically escaping" does not seem to be the right wording according to your comment in https://github.com/symfony/symfony/issues/27796#issuecomment-401703541 😄

Comment (nicolas-grekas) [reply components/process.rst]: reworded :)

**Comment by xabbuh** [comment]
Thank you @nicolas-grekas.

File: components/process.rst [hunk 4]
```diff
@@ -100,6 +100,38 @@ with a non-zero code)::
...
+        The ability to define commands as arrays of arguments was introduced in
+        Symfony 3.3.
+
+    Using array of arguments is the recommended way to define commands. This
+    saves you from any escaping and allows sending signals seamlessly
+    (e.g. to stop processes before completion.)::
+
+        $process = new Process(array('/path/command', '--flag', 'arg 1', 'etc.'));
+
+    If you need use stream redirections, conditional execution, or any other
```
Comment (apfelbox) [inline components/process.rst]: > need to use

Comment (xabbuh) [reply components/process.rst]: thanks, fixed in 2c5579be19da893059b51ff6d01ce4d1ec625c82

**Pull request merged**